### PR TITLE
Greenfield: Manage notifications

### PR DESCRIPTION
**Pull request by dennisreimann**
Prerequisite for btcpayserver/app#12.

### CODE REVIEW

File: BTCPayServer.Client/BTCPayServerClient.Notifications.cs [hunk 1]
```diff
@@ -27,6 +27,21 @@ public partial class BTCPayServerClient
 
             return await HandleResponse<IEnumerable<NotificationData>>(response);
         }
+        
+        public virtual async Task<NotificationSettingsData> GetNotificationSettings(CancellationToken token = default)
+        {
+            var response = await _httpClient.SendAsync(
+                CreateHttpRequest("api/v1/users/me/notification-settings"), token);
```
Comment (Kukks) [inline BTCPayServer.Client/BTCPayServerClient.Notifications.cs:34]:
```suggestion
                CreateHttpRequest("api/v1/users/me/notifications/settings"), token);
```

**Review by Kukks — COMMENTED** [review]
How about notifications/settings instead?

**Comment by dennisreimann** [comment]
There's `notifications/{id}` already and I didn't want to mix it up, but that route would take precedence I guess, so we could change it if you want.

**Review by louieo — COMMENTED** [review]
TACK. Using Postman, I tested getting notification settings and also updating specific and all notification settings. Let me know if you want me to manually test again if you end up changing the routes as per @Kukks suggestion
Screenshot 2024-06-25 at 10 41 45 AM
Screenshot 2024-06-25 at 10 42 48 AM
Screenshot 2024-06-25 at 10 43 06 AM
Screenshot 2024-06-25 at 10 43 22 AM

**Comment by dennisreimann** [comment]
I've also tested this PR inside the app and it works as wanted. 

@Kukks up to you if we want to change the route, otherwise it can be merged.

**Pull request merged**